scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/policy/ExtendedSchedulerPolicy.java

```diff
@@ -121,21 +138,28 @@ public LinkedList<EligibleTaskDescriptor> getOrderedTasks(List<JobDescriptor> jo
                 }
             }
         }
+        schedulingPolicyTimingLogger.end("ESP.getOrderTasks");
+        if (!schedulingPolicyTimingLogger.isHierarchical()) {
+            schedulingPolicyTimingLogger.printTimings(Level.DEBUG);
+            schedulingPolicyTimingLogger.clear();
+        }
         return executionCycleTasks;
     }
 
-    private void scheduleWakeUp(final String taskOrJobId, Date now, Date startAtDate) {
-        if (!startAtCache.containsKey(taskOrJobId)) {
+    private void scheduleWakeUp(Date now, Date startAtDate) {
+        final long scheduledTime = startAtDate.getTime();
+        if (!delayedJobsWakeUpCache.containsKey(scheduledTime)) {
             executor.schedule(() -> {
-                startAtCache.remove(taskOrJobId);
+                delayedJobsWakeUpCache.remove(scheduledTime);
```

Comment:
Not necessary as you are in `if (!delayedJobsWakeUpCache.containsKey(scheduledTime))`

Reply:
hehe, look at the code, it does not do it immediately, it schedules the operation.
When the operation is performed, the key must be removed

Reply:
Oh yes sorry ...